src/Validator/Rules/PossibleFragmentSpreads.php

```diff
@@ -142,14 +142,13 @@ public static function typeIncompatibleAnonSpreadMessage($parentType, $fragType)
     private function getFragmentType(ValidationContext $context, $name)
     {
         $frag = $context->getFragment($name);
-        if ($frag) {
-            $type = TypeInfo::typeFromAST($context->getSchema(), $frag->typeCondition);
-            if ($type instanceof CompositeType) {
-                return $type;
-            }
+        if ($frag === null) {
+            return null;
```

Comment:
A test to cover this case would be great.

Reply:
Will add though this is covered by types (improved in b2d560e2af9cd7512e326005079264ffa447c41e)

Reply:
Looking at it I don't think it's currently possible to invoke that line through our code in src.

Reply:
If there is some invariant making this state impossible, we can add a commented type assertion and remove the condition.

Reply:
What is the way to do it here? `assert` is usually not evaluated in runtime environments. In my projects I use runtime assertion libs. Do we have an exception for it?

Reply:
We are using `assert` in exactly one place as of now. I would go with a PHPDoc to override the type if I know it can not ever be `null` because of previously executed code.

Reply:
Well, it can be executed but one must try very hard and won't happen when using the typical flow.

Either overwriting types using PHPDoc is wrong as it might rot over time. I'm using psalm to prevent me doing it.

TBH I'd keep it as is. It's type covered and method consumers handle `null` value. `getFragmentType()` can return `null` in theory. In practice it won't.